hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -3068,7 +3069,22 @@ public NameQualifierSupport getNameQualifierSupport() {
 	protected final BatchLoadSizingStrategy STANDARD_DEFAULT_BATCH_LOAD_SIZING_STRATEGY = new BatchLoadSizingStrategy() {
 		@Override
 		public int determineOptimalBatchLoadSize(int numberOfKeyColumns, int numberOfKeys) {
-			return 50;
+			int paddedSize = MathHelper.ceilingPowerOfTwo( numberOfKeys );
```

Comment:
IMO it's fine to always do the padding here, and we should also enabled padding (`org.hibernate.cfg.AvailableSettings#IN_CLAUSE_PARAMETER_PADDING`) by default in 6.0, but don't you think this should be controlled by the setting? Or is the setting for IN predicates only so it doesn't need to be respected?

Reply:
probably we have to check the setting

Reply:
done